src/js/services/mopidy/middleware.js

```diff
@@ -1862,30 +1873,75 @@ const MopidyMiddleware = (function () {
               });
           }
 
-          if (subdirectoryImagesToLoad.length) {
-            console.info(`Loading ${subdirectoryImagesToLoad.length} subdirectory URIs`);
-            request(store, 'library.getImages', { uris: subdirectoryImagesToLoad })
-              .then((response) => {
-                const subdirectoriesWithImages = subdirectories.map((subdir) => {
-                  let images = response[subdir.uri] || undefined;
-                  if (images) {
-                    images = formatImages(digestMopidyImages(store.getState().mopidy, images));
-                  }
-                  return {
-                    ...subdir,
-                    images,
-                  };
+          const imagesLoaded = new Promise((resolve) => {
+            if (subdirectoryImagesToLoad.length) {
+              console.info(`Loading ${subdirectoryImagesToLoad.length} subdirectory URIs`);
+              request(store, 'library.getImages', { uris: subdirectoryImagesToLoad })
+                .then((response) => {
+                  resolve(response);
                 });
+            } else {
+              resolve({});
+            }
+          });
 
-                store.dispatch({
-                  type: 'MOPIDY_DIRECTORY_LOADED',
-                  directory: {
-                    uri,
-                    subdirectories: subdirectoriesWithImages,
-                  },
-                });
+          const playlistsLoaded = new Promise((resolve) => {

```

Comment:
This may cause really heavy, un-cancellable requests for slow backends or when the directory contains a lot of playlists. Depending on the backend provider, but this could result in the fetching of the entire playlist including its tracks.

Have you tested it against the likes of Mopidy-Ytmusic or Mopidy-Tidal?